### PR TITLE
Features/multiple packages lists

### PR DESCRIPTION
**Pull request by cholin**
This PR is open for comments. It includes the following changes:

* Initial support for multiple packages lists (see #39)
* add basic packages list for backbone nodes (if there is no 4MB suffix pacakge list file it will use the original list for all profiles)

The resulting directory structure for firmwares looks like the following:

```
firmwares/
  bbb/
     TARGET/
       ImageBuilder
       Images
  vpn/
     TARGET/
       ImageBuilder
       Images
  packages/
    TARGET/
       base/
       luci/
       packages/
       packages_berlin/
       routing/
```

These changes do as well affect our buildbot infrastructure.

### CODE REVIEW

File: packages/bbb.txt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+# Common
+mtr
+iperf
+tmux
+vnstat
+ethtool
+dnsmasq
+iwinfo
+libiwinfo-lua
```
Comment (booo) [inline packages/bbb.txt]: I suggest we add tcpdump here.

Comment (cholin) [inline packages/bbb.txt]: done

**Comment by booo** [comment]
We should add some documentation to the README.

I have to take a closer look but why do we have the image builder twice per target? We only have to create different images/package sets with the image builder per target, right?

**Comment by cholin** [comment]
Well we could as well create a directory for these image builders like for `packages`. But I think for clearness I just adapted the old directory structure (with IB inside the dir with the images). It is as well the same layout which OpenWRT itself uses.

**Comment by andrenarchy** [comment]
Just talked to @cholin and we think it's a good idea to change it to the following layout:
```
firmwares/
  TARGET/
    BUILDBOT_NUMBER/      // only for unstable builds
      OpenWrt-ImageBuilder-....tar.bz2
      bbb/
         Images
      vpn/
         Images
      packages/
         base/
         luci/
         packages/
         packages_berlin/
         routing/
```
Opinions?

**Comment by cholin** [comment]
@andrenarchy @booo can you test this PR and give some feedback? :smile:

**Comment by andrenarchy** [comment]
Testing right now. Regarding the discussion in #212 and #144, I suggest we rename `vpn` to `default` (or some other neutral term).

**Comment by cholin** [comment]
Maybe we should rename both packages list in the following matter:

* `vpn` => `default`
* `bbb` => `minimal`

What do you think?

**Comment by cholin** [comment]
As @booo proposed: Rebased on top of https://github.com/freifunk-berlin/firmware/pull/211.

**Comment by andrenarchy** [comment]
btw: the build failed due to source files that could not be downloaded (see [ar71xx build 479](http://buildbot.berlin.freifunk.net/builders/ar71xx/builds/479)). Maybe the rebase fixed it. If [ar71xx build 481](http://buildbot.berlin.freifunk.net/builders/ar71xx/builds/481) succeeds, I'll also test it locally again.

**Comment by booo** [comment]
I find the name "minimal" misleading. I wouldn't expect for example these packages in a minimal set:
https://github.com/freifunk-berlin/firmware/pull/207/files#diff-37b55f51476a6204d626944e1769b02dR18

"bbb" was fine to me. "expert" would be one suggestions but that's strange too.

**Comment by andrenarchy** [comment]
@booo: yeah, `minimal` may be misleading. But the abbreviation `bbb` may be unclear. `expert` or `advanced` could be an alternative...

**Comment by cholin** [comment]
[ar71xx build 481](http://buildbot.berlin.freifunk.net/builders/ar71xx/builds/481) finished successfully. See [buildbot/unstable/ar71xx/481](http://buildbot.berlin.freifunk.net/buildbot/unstable/ar71xx/481/) for the directory layout.

I agree that `minimal` may be the wrong label. Maybe just `backbone` or `manual`? With `expert` and `advanced` some may expect additional features but quite the contrary is true (there is no wizard at all...).

**Comment by andrenarchy** [comment]
Compilation also works fine here and results in the intended directory structure.

I suggest we merge this PR. Note: it contains the changes from #211 which have been tested by @booo. So #211 can be closed after merging this one.

**Comment by booo** [comment]
I suggest that we document the structure in the README

I vote for backbone as name for "minimal".

We still need to update the buildbot configuration.

**Comment by cholin** [comment]
Added some docu and renamed `minimal` to `backbone`.

**Comment by andrenarchy** [comment]
What exactly needs to be changed in the buildbot config? The latest commit builds and is uploaded to the download location:

 * [ar71xx](http://buildbot.berlin.freifunk.net/buildbot/unstable/ar71xx/486/)
 * [ar71xx_mikrotik](http://buildbot.berlin.freifunk.net/buildbot/unstable/ar71xx_mikrotik/429/)
 * [mpc85xx](http://buildbot.berlin.freifunk.net/buildbot/unstable/mpc85xx/437/)
 * [x86](http://buildbot.berlin.freifunk.net/buildbot/unstable/x86/325/)

**Comment by cholin** [comment]
@booo can you say something about buildbot? I think we could merge this anyhow now :smile:

**Comment by sarumpaet** [comment]
For the record, b1fd7dc1ad17cf581e567bb4ec433fd9abb1c415 fixes a typo in 066e80e3523b0cdc2d47bf843c8d5dd8ff0980c2 that prevented suffixed package lists from being found/built.

**Pull request merged**